src/test/java/org/opensearch/knn/index/NestedSearchIT.java

```diff
@@ -0,0 +1,202 @@
+import static org.opensearch.knn.common.KNNConstants.KNN_METHOD;
+import static org.opensearch.knn.common.KNNConstants.METHOD_HNSW;
+import static org.opensearch.knn.common.KNNConstants.METHOD_PARAMETER_EF_CONSTRUCTION;
+import static org.opensearch.knn.common.KNNConstants.METHOD_PARAMETER_M;
+import static org.opensearch.knn.common.KNNConstants.METHOD_PARAMETER_SPACE_TYPE;
+import static org.opensearch.knn.common.KNNConstants.NAME;
+import static org.opensearch.knn.common.KNNConstants.PARAMETERS;
+import static org.opensearch.knn.common.KNNConstants.PATH;
+import static org.opensearch.knn.common.KNNConstants.QUERY;
+import static org.opensearch.knn.common.KNNConstants.TYPE;
+import static org.opensearch.knn.common.KNNConstants.TYPE_KNN_VECTOR;
+import static org.opensearch.knn.common.KNNConstants.TYPE_NESTED;
+import static org.opensearch.knn.common.KNNConstants.VECTOR;
+
+public class NestedSearchIT extends KNNRestTestCase {
+    private static final String INDEX_NAME = "test-index-nested-search";
+    private static final String FIELD_NAME_NESTED = "test-nested";
+    private static final String FIELD_NAME_VECTOR = "test-vector";
+    private static final String PROPERTIES_FIELD = "properties";
+    private static final int EF_CONSTRUCTION = 128;
+    private static final int M = 16;
+    private static final SpaceType SPACE_TYPE = SpaceType.L2;
+
+    @After
+    @SneakyThrows
+    public final void cleanUp() {
+        deleteKNNIndex(INDEX_NAME);
+    }
+
+    @SneakyThrows
+    public void testNestedSearch_whenKIsTwo_thenReturnTwoResults() {
+        createKnnIndex(2, KNNEngine.LUCENE.getName());
+
+        String doc1 = NestedKnnDocBuilder.create(FIELD_NAME_NESTED)
+            .add(FIELD_NAME_VECTOR, new Float[]{1f, 1f}, new Float[]{1f, 1f})
+            .build();
+        addNestedKnnDoc(INDEX_NAME, "1", doc1);
+
+        String doc2 = NestedKnnDocBuilder.create(FIELD_NAME_NESTED)
+            .add(FIELD_NAME_VECTOR, new Float[]{2f, 2f}, new Float[]{2f, 2f})
+            .build();
+        addNestedKnnDoc(INDEX_NAME, "2", doc2);
+
+        Float[] queryVector = { 1f, 1f };
+        Response response = queryNestedField(INDEX_NAME, 2, queryVector);
+
+        List<Object> hits = (List<Object>) ((Map<String, Object>) createParser(
+            MediaTypeRegistry.getDefaultMediaType().xContent(),
+            EntityUtils.toString(response.getEntity())
+        ).map().get("hits")).get("hits");
+        assertEquals(2, hits.size());
+    }
+
+    /**
+     * {
+     * 		"properties": {
+     * 			"test-nested": {
+     * 				"type": "nested",
+     * 				"properties": {
+     * 					"test-vector": {
+     * 						"type": "knn_vector",
+     * 						"dimension": 3,
+     * 						"method": {
+     * 							"name": "hnsw",
+     * 							"space_type": "l2",
+     * 							"engine": "lucene",
+     * 							"parameters": {
+     * 								"ef_construction": 128,
+     * 								"m": 24
+     *                          }
```

Comment:
nit: please fix formatting for request, not sure if that's the github source viewer but is hard to read

Reply:
Will try to replace from tab to space